fondant_deps/src/lib.rs

```diff
@@ -42,6 +44,9 @@ pub enum FondantError {
 
 /// Derive this trait on a struct to mark it as a 'configuration' struct.
 pub trait Configure: Serialize + DeserializeOwned + Default {
+    fn load_file(config_file: &PathBuf) -> Result<Self, FondantError>;
```

Comment:
perhaps these signatures could be converted to allow more flexibility?
```suggestion
    fn load_file<P: AsRef<Path>>(config_file: P) -> Result<Self, FondantError>;
```